compiler/src/dotty/tools/dotc/quoted/Interpreter.scala

```diff
@@ -0,0 +1,368 @@
+package dotty.tools.dotc
+package quoted
+
+import scala.language.unsafeNulls
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+import java.io.{PrintWriter, StringWriter}
+import java.lang.reflect.{InvocationTargetException, Method => JLRMethod}
+
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.ast.TreeMapWithImplicits
+import dotty.tools.dotc.core.Annotations._
+import dotty.tools.dotc.core.Constants._
+import dotty.tools.dotc.core.Contexts._
+import dotty.tools.dotc.core.Decorators._
+import dotty.tools.dotc.core.Denotations.staticRef
+import dotty.tools.dotc.core.Flags._
+import dotty.tools.dotc.core.NameKinds.FlatName
+import dotty.tools.dotc.core.Names._
+import dotty.tools.dotc.core.StagingContext._
+import dotty.tools.dotc.core.StdNames._
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.core.TypeErasure
+import dotty.tools.dotc.core.Types._
+import dotty.tools.dotc.quoted._
+import dotty.tools.dotc.transform.TreeMapWithStages._
+import dotty.tools.dotc.typer.ImportInfo.withRootImports
+import dotty.tools.dotc.util.SrcPos
+import dotty.tools.repl.AbstractFileClassLoader
+
+
+/** List of classes of the parameters of the signature of `sym` */
+abstract class Interpreter(pos: SrcPos, classLoader: ClassLoader)(using Context):
+  import Interpreter._
+  import tpd._
+
+  type Env = Map[Symbol, Object]
+
+  /** Returns the interpreted result of interpreting the code a call to the symbol with default arguments.
```

Comment:
Looks like this comment was here before but I'm having trouble interpreting it myself :). Is there a word missing around "the code a call",  should it be "the code of a call" ?

Reply:
I rewrote the sentence.